web/source/osk/oskKey.ts

```diff
@@ -47,17 +47,17 @@ namespace com.keyman.osk {
     elementID?: string;
 
     text?: string;
-    sp?: number | keyboards.ButtonClass;
-    width: string;
+    sp?: keyboards.ButtonClass;
+    width: number;
     layer?: string; // The key will derive its base modifiers from this property - may not equal the layer on which it is displayed.
     nextlayer?: string;
-    pad?: string;
+    pad?: number;
     sk?: OSKKeySpec[];
 
-    constructor(id: string, text?: string, width?: string, sp?: number | keyboards.ButtonClass, nextlayer?: string, pad?: string) {
+    constructor(id: string, text?: string, width?: number, sp?: keyboards.ButtonClass, nextlayer?: string, pad?: number) {
       this.id = id;
       this.text = text;
-      this.width = width ? width : "50";
+      this.width = width ? width : 50;
```

Comment:
Can we go ahead and clean up this magic into a constant as we touch it?

Reply:
I thought about it, then remembered this:

https://github.com/keymanapp/keyman/blob/527aaa6618e77acf1e37e51ea765af7ab5e4fb67/common/web/keyboard-processor/src/keyboards/activeLayout.ts#L17-L20

So, _a_ constant for this already exists - `DEFAULT_KEY_WIDTH` - but it doesn't match the value used there.  (Which begs the question:  why is that?  Probably an accident.  _Probably._)

I almost swapped it before submitting, but I'm not sure if replacing it with this different-valued constant would break things or not.  I figured it's better to leave that out of this PR and address it separately so that any breakages caused will be isolated to that change, thus easier to resolve should they occur.

I do think that it's probably a safe replacement - the layout processing layer shouldn't leave any zero-width keys, so the "if false" part of the trinary should never trigger.  But, it's generally better to be safe about this and test appropriately.